nnef/src/deser.rs

```diff
@@ -566,7 +565,9 @@ impl CoerceFrom<Value> for TDim {
         match from {
             Value::Dim(d) => Ok(d.clone()),
             Value::Tensor(t) => Ok(t.to_scalar::<TDim>()?.clone()),
-            Value::Wire(_) => Ok(from.to::<Arc<Tensor>>(builder)?.to_scalar::<TDim>()?.clone()),
+            Value::Wire(_) => {
+                Ok(from.to::<Arc<Tensor>>(builder)?.cast_to::<TDim>()?.to_scalar::<TDim>()?.clone())
```

Comment:
@kali was it what you had in mind when we discussed casting `TDim` and referencing this code section ?

Reply:
Yes!